code/modules/mob/living/basic/jungle/seedling/seedling.dm

```diff
@@ -0,0 +1,316 @@
+	desc = "This oversized, predatory flower conceals what can only be described as an organic energy cannon."
+	icon = 'icons/mob/simple/jungle/seedling.dmi'
+	icon_state = "seedling"
+	icon_living = "seedling"
+	icon_dead = "seedling_dead"
+	habitable_atmos = list("min_oxy" = 2, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minimum_survivable_temperature = 0
+	maximum_survivable_temperature = 450
+	mob_biotypes = MOB_ORGANIC | MOB_PLANT
+	maxHealth = 100
+	health = 100
+	pixel_y = -14
+	base_pixel_y = -14
+	pixel_x = -14
+	base_pixel_x = -14
+	response_harm_continuous = "strikes"
+	response_harm_simple = "strike"
+	melee_damage_lower = 30
+	melee_damage_upper = 30
+	lighting_cutoff_green = 20
+	lighting_cutoff_blue = 25
+	mob_size = MOB_SIZE_LARGE
+	attack_sound = 'sound/weapons/bladeslice.ogg'
+	attack_vis_effect = ATTACK_EFFECT_SLASH
+	ai_controller = /datum/ai_controller/basic_controller/seedling
+	///the state of combat we are in
+	var/combatant_state = SEEDLING_STATE_NEUTRAL
+	///the colors our petals can have
+	var/list/possible_colors = list(COLOR_RED, COLOR_YELLOW, COLOR_OLIVE, COLOR_CYAN)
+	///appearance when we are in our normal state
+	var/mutable_appearance/petal_neutral
+	///appearance when we are in our warmup state
+	var/mutable_appearance/petal_warmup
+	///appearance when we are in the firing state
+	var/mutable_appearance/petal_active
+	///appearance when we are dead
+	var/mutable_appearance/petal_dead
+	///the bucket we carry
+	var/obj/item/reagent_containers/cup/held_can
+	///commands we follow
+	var/list/seedling_commands = list(
+		/datum/pet_command/idle,
+		/datum/pet_command/free,
+		/datum/pet_command/follow,
+	)
+
+/mob/living/basic/seedling/Initialize(mapload)
+	. = ..()
+	var/datum/action/cooldown/mob_cooldown/projectile_attack/rapid_fire/seedling/seed_attack = new(src)
+	seed_attack.Grant(src)
+	ai_controller.set_blackboard_key(BB_RAPIDSEEDS_ABILITY, seed_attack)
+	var/datum/action/cooldown/mob_cooldown/solarbeam/beam_attack = new(src)
+	beam_attack.Grant(src)
+	ai_controller.set_blackboard_key(BB_SOLARBEAM_ABILITY, beam_attack)
+
+	var/petal_color = pick(possible_colors)
+
+	petal_neutral = mutable_appearance(icon, "[icon_state]_overlay")
+	petal_neutral.color = petal_color
+
+	petal_warmup = mutable_appearance(icon, "[icon_state]_charging_overlay")
+	petal_warmup.color = petal_color
+
+	petal_active = mutable_appearance(icon, "[icon_state]_fire_overlay")
+	petal_active.color = petal_color
+
+	petal_dead = mutable_appearance(icon, "[icon_state]_dead_overlay")
+	petal_dead.color = petal_color
+
+	AddElement(/datum/element/wall_smasher)
+	AddComponent(/datum/component/obeys_commands, seedling_commands)
+	RegisterSignal(src, COMSIG_HOSTILE_PRE_ATTACKINGTARGET, PROC_REF(pre_attack))
+	RegisterSignal(src, COMSIG_KB_MOB_DROPITEM_DOWN, PROC_REF(drop_can))
+	update_appearance()
+
+
+/mob/living/basic/seedling/proc/assign_tamer(mob/living/tamer)
+	befriend(tamer)
```

Comment:
oh wait i'm stupid lol, just call `befriend()` directly in the `/obj/item/seeds/seedling` code, no need to do it like this especially since proc overhead is a thing